consensus/replay.go

```diff
@@ -324,7 +324,12 @@ func (h *Handshaker) ReplayBlocks(
 			}
 
 			if res.ConsensusParams != nil {
-				state.ConsensusParams = types.PB2TM.ConsensusParams(res.ConsensusParams)
+				// Preserve TimeIotaMs since it's not exposed to the application.
+				timeIotaMs := state.ConsensusParams.Block.TimeIotaMs
+				{
+					state.ConsensusParams = types.PB2TM.ConsensusParams(res.ConsensusParams)
```

Comment:
I feel like a better approach would be to build this into the PB2TM.ConsensusParams method. Then we know we'll never forget to do this

Reply:
Openned #3432